sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/UnsafeArrayData.java

```diff
@@ -227,74 +226,45 @@ public Decimal getDecimal(int ordinal, int precision, int scale) {
     if (precision <= Decimal.MAX_LONG_DIGITS()) {
       return Decimal.apply(getLong(ordinal), precision, scale);
```

Comment:
Other classes use `return Decimal.createUnsafe(getLong(ordinal), precision, scale);` for same condition. Maybe this is just a missing spot, or some reason to not apply `createUnsafe`?